huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset.go

```diff
@@ -0,0 +1,598 @@
+	if err != nil {
+		return common.CheckDeletedDiag(d, parseDatasetErrorToError404(err), "Error retrieving ModelArts dataset")
+	}
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("name", detail.DatasetName),
+		d.Set("type", detail.DatasetType),
+		d.Set("data_format", detail.DataFormat),
+		d.Set("output_path", detail.WorkPath),
+		d.Set("description", detail.Description),
+		setDataSourcesToState(d, detail.DataSources[0]),
+		setSchemaToState(d, detail.Schema),
+		d.Set("import_labeled_enabled", detail.ImportData),
+		setLabelsToState(d, detail.Labels),
+		d.Set("created_at", utils.FormatTimeStampUTC(int64(detail.CreateTime)/1000)),
+		d.Set("status", detail.Status),
+	)
+
+	if mErr.ErrorOrNil() != nil {
+		return fmtp.DiagErrorf("Error setting ModelArts dataset fields: %s", mErr)
+	}
+
+	return nil
+}
+
+func resourceDatasetUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.ModelArtsV2Client(region)
+	if err != nil {
+		return diag.Errorf("Error creating ModelArts v2 client, err=%s", err)
+	}
+
+	_, err = dataset.Get(client, d.Id(), dataset.GetOpts{})
+	if err != nil {
+		return common.CheckDeletedDiag(d, parseDatasetErrorToError404(err), "Error retrieving ModelArts dataset")
+	}
+
+	desc := d.Get("description").(string)
+	updateParams := dataset.UpdateOpts{
+		DatasetName: d.Get("name").(string),
+		Description: &desc,
+	}
+
+	if d.HasChange("labels") {
+		o, n := d.GetChange("labels")
+		updateParams.AddLabels = buildLabelsParamter(n)
+		updateParams.DeleteLabels = buildLabelsParamter(o)
+	}
+
+	rst := dataset.Update(client, d.Id(), updateParams)
+	if rst.Err != nil {
+		return fmtp.DiagErrorf("Update ModelArts dataset=%s failed, error: %s", d.Id(), err)
+	}
+
+	return resourceDatasetRead(ctx, d, meta)
+}
+
+func resourceDatasetDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.ModelArtsV2Client(region)
+	if err != nil {
+		return diag.Errorf("Error creating ModelArts v2 client, err=%s", err)
+	}
+
+	dErr := dataset.Delete(client, d.Id())
+	if dErr.Err != nil {
+		return common.CheckDeletedDiag(d, parseDatasetErrorToError404(err), "Delete ModelArts dataset failed.")
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func waitingforDatasetCreated(ctx context.Context, client *golangsdk.ServiceClient, id string,
+	timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"0", "5", "6", "7", "8"},
+		Target:  []string{"1"},
+		Refresh: func() (interface{}, string, error) {
+
+			resp, err := dataset.Get(client, id, dataset.GetOpts{})
+			if err != nil {
+				return nil, "", err
+			}
+			return resp, fmt.Sprint(resp.Status), nil
+		},
+		Timeout:      timeout,
+		PollInterval: 20 * timeout,
```

Comment:
it seems that `PollInterval` is a huge value

Reply:
fix